speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/FileDataStorageTests.cpp

```diff
@@ -4,26 +4,72 @@
+                                        timeBlackBox,                     \
+                                        "environmentAlsoFake.xml",        \
+                                        "Name of a dummy environment",    \
+                                        0,                                \
                                         INTERVENTION_TYPE_ONLY_SIGNALS
 
+/// @brief Get an unsigned long variable "tickCount" with a random value
+#define GET_RANDOM_TICKCOUNT unsigned long tickCount = random.NextUInt();
+
+/// @brief Get a tCarElt variable "car" and fill it with random data for variables relevant to the FileDataStorage system.
+#define GET_RANDOM_CAR                          \
+    tCarElt car = {};                           \
+    car.pub.DynGCg.pos.x = random.NextFloat();  \
+    car.pub.DynGCg.pos.y = random.NextFloat();  \
+    car.pub.DynGCg.pos.z = random.NextFloat();  \
+    car.pub.DynGCg.pos.ax = random.NextFloat(); \
+    car.pub.DynGCg.pos.ay = random.NextFloat(); \
+    car.pub.DynGCg.pos.az = random.NextFloat(); \
+    car.pub.DynGC.vel.x = random.NextFloat();   \
+    car.pub.DynGC.acc.x = random.NextFloat();   \
+    car.priv.gear = random.NextInt();           \
+    car.ctrl.steer = random.NextFloat();        \
+    car.ctrl.brakeCmd = random.NextFloat();     \
+    car.ctrl.accelCmd = random.NextFloat();     \
+    car.ctrl.clutchCmd = random.NextFloat();
+
+/// @brief Get a tSituation variable "situation" and fill it with random data for variables relevant to the FileDataStorage system.
+#define GET_RANDOM_SITUATION tSituation situation = {};  // Currently no variables of tSituation are used yet
+
+/// @brief Write to a stream variable "expected" the data expected to be seen in a file when environment is stored
+#define WRITE_EXPECTED_ENVIRONMENT expected << std::to_string(car.pub.DynGCg.pos.x) << "\n"  \
+                                            << std::to_string(car.pub.DynGCg.pos.y) << "\n"  \
+                                            << std::to_string(car.pub.DynGCg.pos.z) << "\n"  \
+                                            << std::to_string(car.pub.DynGCg.pos.ax) << "\n" \
+                                            << std::to_string(car.pub.DynGCg.pos.ay) << "\n" \
+                                            << std::to_string(car.pub.DynGCg.pos.az) << "\n" \
+                                            << std::to_string(car.pub.DynGC.vel.x) << "\n"   \
+                                            << std::to_string(car.pub.DynGC.acc.x) << "\n"   \
+                                            << std::to_string(car.priv.gear) << "\n";
+
+#define WRITE_EXPECTED_CONTROLS expected << std::to_string(car.ctrl.steer) << "\n"    \
+                                         << std::to_string(car.ctrl.brakeCmd) << "\n" \
+                                         << std::to_string(car.ctrl.accelCmd) << "\n" \
+                                         << std::to_string(car.ctrl.clutchCmd) << "\n"
```

Comment:
Change all '\n' characters with std::endl as enters are different on windows and linux